SwanCustomEnvironments/swancustomenvironments/templates/customenvs.html

```diff
@@ -0,0 +1,88 @@
+{% extends "page.html" %}
+
+{% block stylesheet %}
+{{super()}}
+<style type="text/css">
+/* disable initial hide */
+div#header, div#site {
+    display: block;
+}
+#output-box {
+	display: block;
+	width: 1150px;
+	height: 300px;
+	border: 1px solid #ccc;
+	background-color: #f7f7f7;
+	padding: 10px;
+	border-radius: 5px;
+}
+</style>
+{% endblock %}
+
+{% block params %}
+{{super()}}
+data-base-url="{{base_url | urlencode}}"
+{% endblock %}
+
+{% block site %}
+
+<div class="container">
+  <div class="row">
+    <div class="swan-info">
+      <div id="swan-loader">
+        <div class="loader-circle">
+          <img src="{{hub_prefix}}static/swan/logos/{{swan_logo_filename}}">
+        </div>
+        <div class="loader-line-mask">
+          <div class="loader-line"></div>
+        </div>
+        <span class="text">Creating environment...</span>
+		<div class="swan-info" id="output-box"></div>
+      </div>
+    </div>
+  </div>
+</div>
+
+{% endblock %}
+
+{% block script %}
+    {{super()}}
+
+<script type = "text/javascript" >
+	
+	function error(reason) {
+		var loader = document.getElementById('swan-loader');
+		loader.getElementsByClassName('loader-line-mask')[0].style.display = 'none';
+		loader.getElementsByClassName('text')[0].innerHTML = `<p class=\"extra\">Error downloading Project: ${reason}</p>`;
+	}
+
+	var base_url = document.body.getAttribute('data-base-url');
+	
+	var env_name = encodeURIComponent(document.body.getAttribute('env') || "");
+	var req = encodeURIComponent(document.body.getAttribute('req') || "");
+	var clear = encodeURIComponent(document.body.getAttribute('clear') || "true");
+	var accpy_version = encodeURIComponent(document.body.getAttribute('accpy') || "");
+	var custom_python = encodeURIComponent(document.body.getAttribute('python') || "");
+
+	var eventSource = new EventSource(`${base_url}api/customenvs?env=${env_name}&req=${req}&clear=${clear}&accpy=${accpy_version}&python=${custom_python}`);
```

Comment:
So here you create a source of events which does an initial GET of api/customenvs, and later you register a listener of events that come back from that channel? And the other end if the SwanCustomEnvironmentsApiHandler, which when uses its `write` method, it writes a message on that same channel?

Reply:
Yup, that's it. You got it right. I found this ilustration on the internet to it make more illustrative
![image](https://github.com/swan-cern/jupyter-extensions/assets/54753829/bd5efa72-5423-4c87-b4ee-754edb205575)